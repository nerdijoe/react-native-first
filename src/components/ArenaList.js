import React from 'react'
import { connect } from 'react-redux'

import {
  View,
  Image,
  TouchableOpacity,
  FlatList,
  ScrollView
} from 'react-native'
import { Actions } from 'react-native-router-flux'

const styles = {
  cardImage: {
    width: 518 / 3,
    height: 539 / 3
  },
  container: {
    display: 'flex',
    padding: 10,
    marginTop: 50,
    marginBottom: 30,
    flexWrap: 'wrap',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  card: {
    backgroundColor: 'white',
    flexDirection: 'column',
    margin: 5,
    borderRadius: 8,
    borderStyle: 'solid',
    borderWidth: 0,
    alignItems: 'center'

  },
  cardDesc: {
    fontSize: 15,
    padding: 10
  }
}

class ArenaList extends React.Component {
  onPressHandler (arena) {
    console.log('onPressHandler card_id=' + arena._id)
    Actions.adetail({arena: arena})
  }

  render () {
    return (
      <View style={{flex: 1}} >
        <ScrollView >
          <View style={styles.container}>
            {this.props.arenas.map(arena => {
              return (
                <TouchableOpacity key={arena._id} style={styles.card} onPress={() => { this.onPressHandler(arena) }}>
                  <Image
                    style={styles.cardImage}
                    source={{uri: `http://www.clashapi.xyz/images/arenas/${arena.idName}.png`}}
                  />
                  <Text style={styles.cardDesc}>{arena.name}</Text>
                </TouchableOpacity>
              )
            })}
          </View>
        </ScrollView>
      </View>

    )
  }
}

const mapStateToProps = (state) => {
  return {
    arenas: state.arenaReducer.data
  }
}

const connectedArenaList = connect(mapStateToProps, null)(ArenaList)
export default connectedArenaList
