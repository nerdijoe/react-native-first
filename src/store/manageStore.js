import { createStore, applyMiddleware, compose } from 'redux'
import logger from 'redux-logger'
import thunk from 'redux-thunk'

import clasApp from '../reducers'

const middlewares = applyMiddleware( logger, thunk )

const store = createStore(clasApp, middlewares)

export default store
