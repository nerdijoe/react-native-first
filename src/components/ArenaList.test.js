import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import ArenaList from './ArenaList'
import store from '../store/manageStore'

describe('ArenaList', () => {

  it('renders without crashing', ()=> {
    const tree = renderer.create(<Provider store={store}><ArenaList /></Provider>)
    // const tree = renderer.create(<CardList cards={cards}/>)
    expect(tree).toMatchSnapshot()
  })
})
