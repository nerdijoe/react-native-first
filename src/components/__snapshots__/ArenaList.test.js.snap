// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArenaList renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "marginBottom": 30,
            "marginTop": 50,
            "padding": 10,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flexDirection": "column",
              "margin": 5,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={
              Object {
                "uri": "http://www.clashapi.xyz/images/arenas/training-camp.png",
              }
            }
            style={
              Object {
                "height": 179.66666666666666,
                "width": 172.66666666666666,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 15,
                "padding": 10,
              }
            }
          >
            Training Camp
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flexDirection": "column",
              "margin": 5,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={
              Object {
                "uri": "http://www.clashapi.xyz/images/arenas/goblin-stadium.png",
              }
            }
            style={
              Object {
                "height": 179.66666666666666,
                "width": 172.66666666666666,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 15,
                "padding": 10,
              }
            }
          >
            Goblin Stadium
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flexDirection": "column",
              "margin": 5,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={
              Object {
                "uri": "http://www.clashapi.xyz/images/arenas/bone-pit.png",
              }
            }
            style={
              Object {
                "height": 179.66666666666666,
                "width": 172.66666666666666,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 15,
                "padding": 10,
              }
            }
          >
            Bone Pit
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flexDirection": "column",
              "margin": 5,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={
              Object {
                "uri": "http://www.clashapi.xyz/images/arenas/barbarian-bowl.png",
              }
            }
            style={
              Object {
                "height": 179.66666666666666,
                "width": 172.66666666666666,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 15,
                "padding": 10,
              }
            }
          >
            Barbarian Bowl
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
